File: features/structures/enumerations.cpp
enum Rank {
    jack, queen, king, ace = 1, two = 2, three = 3, four = 4, five = 5, six=6, seven=7, eight=8, nine=9, ten=10
};

string simpleDescription(Rank r) {
        switch(r) {
            case ace:
                return "ace"
            case jack:
                return "jack"
            case queen:
                return "queen"
            case king:
                return "king"
            default:
                return to_string((int)r)
        }
    }
}
Rank a = two;
int twoValue = two;

enum Suit {
    spades, hearts, diamonds, clubs
};

string simpleDescription(Suit s) {
        switch(s) {
            case spades:
                return "spades"
            case hearts:
                return "hearts"
            case diamonds:
                return "diamonds"
            case clubs:
                return "clubs"
        }
    }
}
Suit h = hearts;
string heartsDescription = simpleDescription(h);

File: features/structures/enumerations.java



//Java enumerations do not have raw values
public enum Rank {
    ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN,
    JACK, QUEEN, KING
}


public enum Suit {
    SPADES, HEARTS, DIAMONDS, CLUBS
}

public class Main {
	
	public String simpleDescription(Rank r) {
	        switch(r) {
	            case ACE:
	                return "ace";
	            case JACK:
	                return "jack";
	            case QUEEN:
	                return "queen";
	            case KING:
	                return "king";
	            default:
	                return String.parseInt(r);
	        }
	    }
	}
	
	public String simpleDescription(Suit r) {
	        switch(s) {
	            case SPADES = return "spades";
	            case HEARTS = return "hearts";
	            case DIAMONDS = return "diamonds";
	            case CLUBS = return "clubs";
	        }
	    }
	}
	
	public static void main(String[] args) {
			Suit hearts = Suit.HEARTS;
			Rank ace = Rank.ACE;
			
	}
}
